testutil/network/network.go

```diff
@@ -14,33 +14,42 @@ import (
 	"github.com/cosmos/cosmos-sdk/testutil/network"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	"github.com/tendermint/spm/cosmoscmd"
 	tmrand "github.com/tendermint/tendermint/libs/rand"
 	tmdb "github.com/tendermint/tm-db"
 
 	"github.com/ChihuahuaChain/chihuahua/app"
+	"github.com/ChihuahuaChain/chihuahua/testutil"
 )
 
 type (
 	Network = network.Network
 	Config  = network.Config
 )
 
-// New creates instance with fully configured cosmos network.
-// Accepts optional config, that will be used in place of the DefaultConfig() if provided.
-func New(t *testing.T, configs ...network.Config) *network.Network {
-	if len(configs) > 1 {
-		panic("at most one config should be provided")
+// New creates instance with fully configured Chihuahua network, along with
+// creating a funded account for each name provided
+func New(t *testing.T, config network.Config, accounts ...string) (*network.Network, keyring.Keyring) {
+	kr := testutil.GenerateKeyring(t)
+
+	// add genesis accounts
+	genAuthAccs := make([]authtypes.GenesisAccount, len(accounts))
+	genBalances := make([]banktypes.Balance, len(accounts))
+	for i, name := range accounts {
+		a, b := newGenAccout(kr, name, 1000000000000)
+		genAuthAccs[i] = a
+		genBalances[i] = b
```

Comment:
this part is new and not strictly necessary, might delete even though it could be useful in the future